contracts/KNS_Retirer_v1.sol

```diff
@@ -146,38 +146,48 @@ contract KNS_Retirer is Initializable, OwnableUpgradeable {
     
     // PUBLIC
 
-    function retireAndKI( uint _USDCAmt ) public {
+    function retireAndKI(
+        uint _USDCAmt,
+        address beneficiary, 
+        string memory domainName
+    ) public {
         require(
             ( _USDCAmt >= 1*(10**6) ), 
             "Payment net of royalties and referrals is less than 1 USDC."
         );
 
         IERC20(USDC).transferFrom(msg.sender, address(this), _USDCAmt);
-        retireBCT((_USDCAmt * retireBCTPercent) / percentFundsReceived );
-        stakeinKI ((_USDCAmt * stakeInKIPercent) / percentFundsReceived );
+
+        retireBCT(
+            (_USDCAmt * retireBCTPercent) / percentFundsReceived,
+            beneficiary,
+            domainName
+        );
         
-        //Check balance after retirement and allocation to auto-retirement KLIMA.
-        uint256 leftoverFunds = IERC20(USDC).balanceOf(address(this));
+        stakeinKI ((_USDCAmt * stakeInKIPercent) / percentFundsReceived );
         
-        IERC20(USDC).approve(treasurer, leftoverFunds);
```

Comment:
@rajasmasala I think this approval was redundant, since the contract is sending funds that it already owns. So it can do a simple `transfer()` instead. Let me know if I misunderstood something here.

Reply:
Agreed, we can test to be sure all of these changes work in practice but I really appreciate the refactoring. Code is much cleaner now.